frontend/components/widgets/map/component.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { FC, useState, useCallback } from 'react';
+import { LayerManager, Layer } from '@vizzuality/layer-manager-react';
+import PluginMapboxGl from '@vizzuality/layer-manager-plugin-mapboxgl';
+import Map from 'components/map';
+import getBBox from '@turf/bbox';
+import getCentroid from '@turf/centroid';
+
+import type { MapWidgetProps } from './types';
+import { MapEvent, Popup } from 'react-map-gl';
+
+const MapWidget: FC<MapWidgetProps> = ({ featureTooltip, selectedRegion, extraLayers = [] }: MapWidgetProps) => {
+  const [map, setMap] = useState(null);
+  const [viewport, setViewport] = useState({
+    latitude: 54.123389,
+    longitude: -124.950408,
+    zoom: 4,
+    minZoom: 4,
+    maxZoom: 6,
+  });
+  const [bounds, setBounds] = useState({
+    bbox: null,
+    options: {
+      padding: 50,
+    },
+    viewportOptions: {
+      transitionDuration: 0,
+    },
+  });
+  const [highlightedFeature, setHighlightedFeature] = useState(null);
+
+  const handleViewportChange = useCallback((vw) => {
+    setViewport(vw);
+  }, []);
+
+  const handleHover = (evt: MapEvent) => {
+    const feature = evt.features.find((f) => !!f.properties.TOURISM_REGION_NAME);
+    if (feature) {
+      const source = 'tourism_regions';
+      const sourceLayer = 'tourism_regions';
+      const centroid = getCentroid(feature);
+      centroid.properties = feature.properties;
+      centroid.id = feature.id;
+      if (highlightedFeature) {
+        map.setFeatureState({ source, id: highlightedFeature.id, sourceLayer }, { hover: false });
+      }
+      setHighlightedFeature(centroid);
+      map.setFeatureState({ source, id: feature.id, sourceLayer }, { hover: true });
+    } else if (highlightedFeature) {
+      resetHighlight();
+    }
+  };
+  const handleMapMouseLeave = () => {
+    resetHighlight();
+  };
+  const resetHighlight = () => {
+    const layer = 'tourism_regions';
+
+    if (highlightedFeature) {
+      map.setFeatureState({ source: layer, id: highlightedFeature.id, sourceLayer: layer }, { hover: false });
+      setHighlightedFeature(null);
+    }
+  };
+  const handleMapLoad = ({ map }) => {
+    setMap(map);
+    setTimeout(() => {
+      if (selectedRegion) {
+        const selectedRegionFilter = ['match', ['get', 'TOURISM_REGION_NAME'], selectedRegion, true, false];
+        map.setFilter('tourism_regions_outline', selectedRegionFilter);
+        map.setFilter('tourism_regions_fills', selectedRegionFilter);
+
+        setTimeout(() => {
+          const features = map.queryRenderedFeatures({ layers: ['tourism_regions_outline'] });
+          if (features.length > 0) {
+            const zoomToBbox = getBBox(features[0]);
+            setBounds({
+              ...bounds,
+              bbox: zoomToBbox,
+            });
+          }
+        }, 200);
```

Comment:
@Bluesmile82 do you know if there is any event for LayerManager or that Map component that will be emitted after source features are loaded? I could use `querySourceFeatures` instead of rendered but still this needs to be run after data is loaded.

Reply:
I could also use `setInterval` to check if features are already available, that would be a better workaround than just one `setTimeout`.

Reply:
I'm not sure why do you need to find out when the features are loaded. If you have the bounds for each region and you know when it's filtered can't you just fit the bounds directly without waiting for the features to show?

Also, In general, is better to add all the filters directly to a static style. As I think the map will rerender if you use this setFilter.

Reply:
Actually, you could add a useEffect to just change the viewport and adjust the bounds. I think it's better to know the bounds in advance for each region:

```
  useEffect(() => {
        setViewport({
        ...viewport, bounds: REGION_BOUNDS[selectedRegion]
      });
  }, [selectedRegion]);

```

Reply:
So I don't have region_bounds, it's fetched by mapbox component as part of the layer so I want to access the data to get feature geometries.